app/src/main/java/com/mahmoudrashad/websocket/MainActivity.java

```diff
@@ -3,12 +3,122 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import tech.gusavila92.websocketclient.WebSocketClient;
 
 public class MainActivity extends AppCompatActivity {
 
+    private WebSocketClient webSocketClient;
+    TextView textView;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+
+        createWebSocketClient();
+    }
+
+    private void createWebSocketClient() {
+
+        URI uri;
+        try {
+            // Connect to local host
+            uri = new URI("ws://10.0.2.2:8080/websocket");
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+            return;
+        }
+
+
+        webSocketClient = new WebSocketClient(uri) {
+            @Override
+            public void onOpen() {
+
+                Log.i("WebSocket", "Session is starting");
+                webSocketClient.send("Hello World!");
+
+            }
+
+            @Override
+            public void onTextReceived(String s) {
+
+                Log.i("WebSocket", "Message received");
+                final String message = s;
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        try{
+                            TextView textView = findViewById(R.id.T_V);
```

Comment:
please use the clean naming  format
- must have meaning  
-first litter in small letter 
-if content more than one-word male each first letter it large letter